hw_steering_parser.py

```diff
@@ -0,0 +1,97 @@
+#SPDX-License-Identifier: BSD-3-Clause
+#Copyright (c) 2021 NVIDIA CORPORATION. All rights reserved.
+
+import argparse
+import csv
+
+from hw_steering_src.dr_common import *
+from hw_steering_src.dr_context import *
+from hw_steering_src.dr_table import *
+from hw_steering_src.dr_matcher import *
+
+# mapping csv records types to it's relevant parser function
+switch_csv_res_type = {
+    MLX5DR_DEBUG_RES_TYPE_CONTEXT: dr_parse_context,
+    MLX5DR_DEBUG_RES_TYPE_CONTEXT_ATTR: dr_parse_context_attr,
+    MLX5DR_DEBUG_RES_TYPE_CONTEXT_CAPS: dr_parse_context_caps,
+    MLX5DR_DEBUG_RES_TYPE_CONTEXT_SEND_ENGINE: dr_parse_context_send_engine,
+    MLX5DR_DEBUG_RES_TYPE_CONTEXT_SEND_RING: dr_parse_context_send_ring,
+    MLX5DR_DEBUG_RES_TYPE_TABLE: dr_parse_table,
+    MLX5DR_DEBUG_RES_TYPE_MATCHER: dr_parse_matcher,
+    MLX5DR_DEBUG_RES_TYPE_MATCHER_ATTR: dr_parse_matcher_attr,
+    MLX5DR_DEBUG_RES_TYPE_MATCHER_NIC_RX: dr_parse_matcher_nic,
+    MLX5DR_DEBUG_RES_TYPE_MATCHER_NIC_TX: dr_parse_matcher_nic,
+    MLX5DR_DEBUG_RES_TYPE_MATCHER_TEMPLATE: dr_parse_matcher_template,
+#    MLX5DR_DEBUG_RES_TYPE_MATCHER_TEMPLATE_DEFINER: dr_parse_matcher_template_definer,
+}
+
+
+def dr_csv_get_obj(line):
+    res_type = line[0]
+    if res_type not in switch_csv_res_type.keys():
+        return None
+
+    parser = switch_csv_res_type[line[0]]
+    return parser(line)
+
+
+def dr_parse_csv_file(csv_file):
+    ctx = None
+    last_table = None
+    last_matcher = None
+    last_send_engine = None
+    csv_reader = csv.reader(csv_file)
+    for line in csv_reader:
+        obj = dr_csv_get_obj(line)
+        if line[0] == MLX5DR_DEBUG_RES_TYPE_CONTEXT:
+            ctx = obj
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_CONTEXT_ATTR:
+            ctx.add_attr(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_CONTEXT_CAPS:
+            ctx.add_caps(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_CONTEXT_SEND_ENGINE:
+            last_send_engine = obj
+            ctx.add_send_engine(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_CONTEXT_SEND_RING:
+            last_send_engine.add_send_ring(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_TABLE:
+            last_table = obj
+            ctx.add_table(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_MATCHER:
+            last_matcher = obj
+            last_table.add_matcher(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_MATCHER_ATTR:
+            last_matcher.add_attr(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_MATCHER_NIC_RX:
+            last_matcher.add_nic_rx(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_MATCHER_NIC_TX:
+            last_matcher.add_nic_tx(obj)
+        elif line[0] == MLX5DR_DEBUG_RES_TYPE_MATCHER_TEMPLATE:
```

Comment:
sort conditions according to rec type frequency, like  MLX5DR_DEBUG_RES_TYPE_CONTEXT should be last